src/test/java/core/basesyntax/service/impl/ProduceMapReportServiceTest.java

```diff
@@ -0,0 +1,87 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.service.Producer;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class ProduceMapReportServiceTest {
+    private static final String VALID_DATA = "banana,152" + System.lineSeparator() + "apple,90";
+    private static final String EMPTY_DATA = "";
+    private static final String ZERO_QUANTITY_DATA = "banana,0";
+    private static final String SINGLE_FRUIT_DATA = "banana,5";
+    private static final String LARGE_QUANTITY_DATA = "banana,1000"
+            + System.lineSeparator()
+            + "apple,2000";
+    private static Producer produceMapReportService;
+    private String expected;
+
+    @BeforeAll
+    public static void setUp() {
+        produceMapReportService = new ProduceMapReportService();
+    }
+
+    @Test
+    public void produceMapReportService_ValidData_Ok() {
+        expected = VALID_DATA;
+        Map<String, Integer> map = new LinkedHashMap<>();
+        map.put("banana", 152);
+        map.put("apple", 90);
+        String actual = produceMapReportService.produceReport(map);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void produceMapReportService_NullData_NotOk() {
+        assertThrows(IllegalArgumentException.class,
+                () -> produceMapReportService.produceReport(null));
+    }
+
+    @Test
+    public void produceMapReportService_EmptyData_NotOK() {
+        Map<String, Integer> map = Map.of();
+        expected = EMPTY_DATA;
+        String actual = produceMapReportService.produceReport(map);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void produceMapReportService_NegativeQuantity_NotOk() {
+        Map<String, Integer> map = new LinkedHashMap<>();
+        map.put("banana", -5);
+        assertThrows(IllegalArgumentException.class,
+                () -> produceMapReportService.produceReport(map));
+    }
+
+    @Test
+    public void produceMapReportService_ZeroQuantity_Ok() {
+        Map<String, Integer> map = new LinkedHashMap<>();
```

Comment:
You always create a new map in every test, it will be better if you think about initialize it before all tests. Also, if you need to clear the data from the map, think about how you can achieve this.